core/backend/src/IpcHost.ts

```diff
@@ -166,16 +166,28 @@ export abstract class IpcHandler {
 
         return { result: await func.call(impl, ...args) };
       } catch (err: any) {
-        const ret: IpcInvokeReturn = {
-          error: {
-            name: err.hasOwnProperty("name") ? err.name : err.constructor?.name ?? "Unknown Error",
-            message: err.message ?? BentleyError.getErrorMessage(err),
-            errorNumber: err.errorNumber ?? 0,
-          },
-        };
-        if (!IpcHost.noStack)
-          ret.error.stack = BentleyError.getErrorStack(err);
-        return ret;
+        // Returning the ConflictingLocksError class results in information loss due to the way IPC handles error objects, so we get around this by reconstructing on the frontend using the provided
+        // constructorName and args.
+        if (err instanceof ConflictingLocksError) {
+          const ret: IpcInvokeReturn = {
+            errorConstructorName: err.constructor.name,
+            argsForErrorConstructor: [err.message, err.getMetaData(), err.conflictingLocks],
+          };
+          if (!IpcHost.noStack)
+            ret.stack = BentleyError.getErrorStack(err);
+          return ret;
```

Comment:
Can't these three lines (and the identical ones in the else clause) be pulled out of the if/then/else?

Reply:
Never mind; I misread.